src/openapi/streaming/subscription.js

```diff
@@ -351,6 +401,8 @@ Subscription.prototype.reset = function() {
             break;
 
         case STATE_SUBSCRIBED:
+        case STATE_PATCH_REQUESTED:
+            this.onUnsubscribe();

```

Comment:
if a patch is in process and a reset happens, then the handler for the patch will set the state to subscribed when it comes back, even if we are in a subscribe requested.. it just feels a bit fishy here..